edgraph/server.go

```diff
@@ -827,7 +827,11 @@ func buildUpsertQuery(qc *queryContext) string {
 	}
 
 	qc.condVars = make([]string, len(qc.req.Mutations))
-	upsertQuery := strings.TrimSuffix(qc.req.Query, "}")
+
+        // We use String Builder for effective string concatenation
+        var upsertQueryBuilder strings.Builder
```

Comment:
Let's keep the variable name short. Maybe just rename it to `builder` or just `b`. In Go, short local variables are idiomatic. Also can you please check if the indentation of text is fine?

Reply:
Done!